### ISSUE TITLE
Channel connections linger in Eyre state

### ISSUE THREAD

**Issue by Fang-** [urbit/urbit#4617]
Happened to look at debug dashboard just now, and noticed I had 1216 connections open in eyre. Notably I was not running `urbit-king` (see also #4530), instead still on the `urbit` rc-1 that shipped with flagday.

I have since updated to latest vere, but can still reproduce connection creep by simply reloading my landscape page. The channel (in Eyre state) itself gets properly cleaned up, but its connection lingers.

Arguably Eyre should remove the connection from state, as it does for completion or error cases. But I think we managed to get `%cancel-request` tasks for these cases in the past, so I wonder what changed...

**Comment by pilfer-pandex** [comment]
Do you mean to say that vere sends a %cancel-request at the end of every *successful* request, not just when an exception(al situation) occurs?

**Comment by Fang-** [comment]
Not sure if I was saying that or not, but I don't think that's right.

I now suspect this is just a matter of eyre not closing the connection after receiving `%delete` on a channel. But it has me wondering how this wasn't an issue before. Was it just never noticed somehow? Peculiar.

**Comment by Fang-** [comment]
Yes, but only for requests that haven't gotten an initial response yet, which the channels almost certainly are giving.

As described in the opening comment here, I saw this while running the C king.

**Comment by joemfb** [comment]
in `_http_hgen_stop()`, we send `%cancel-request` if the *response* is not done, so this case is supposed to be covered. But it still sounds like there's a gap in the state machine somewhere ...
